yellow-paper/docs/rollup-circuits/index.md

```diff
@@ -0,0 +1,429 @@
+---
+title: Rollup Circuits
+sidebar_position: 99
+---
+
+## Overview
+
+Together with the [validating light node](./../contracts/index.md) the rollup circuits is was must ensure that incoming blocks are valid, that state is progressed correctly and that anyone can rebuild the state.
+
+To support this, we construct a single proof for the entire block, which is then verified by the validating light node. This single proof is constructed by recursively merging proofs together in a binary tree structure. This structure  allows us to keep the workload of the individual proof small, while making it very parallelizable. This works very well for case where we want many actors to be able to participate in the proof generation.
+
+The tree structure is outlined below, but the general idea is that we have a tree where all the leaves are transactions (kernel proofs) and through $\log n$ steps we can then "compress" them down to just a single root proof. Note that we have three (3) different types of "merger" circuits, namely:
+- The base rollup
+  - Merges two kernel proofs
+- The merge rollup
+  - Merges two base rollup proofs OR two merge rollup proofs
+- The root rollup
+  - Merges two merge rollup proofs 
```

Comment:
The root becomes the top, you can take merges from earlier, but it is a bit odd to do since you would just reduce your block size after you already built it 🤷